x/examples/outline-cli/dns_linux.go

```diff
@@ -27,40 +28,77 @@ const (
 	resolvConfHeadBackupFile = "/etc/resolv.head.outlinecli.backup"
 )
 
+// systemDNSBackup info about systemdns backup
+type systemDNSBackup struct {
+	original, backup string
+}
+
+var systemDNSBackups = make([]systemDNSBackup, 0, 2)
+
 func setSystemDNSServer(serverHost string) error {
 	setting := []byte(`# Outline CLI DNS Setting
 # The original file has been renamed as resolv[.head].outlinecli.backup
 nameserver ` + serverHost + "\n")
 
-	if err := backupAndWriteFile(resolvConfFile, resolvConfBackupFile, setting); err != nil {
-		return err
+	err := backupAndWriteFile(resolvConfFile, resolvConfBackupFile, setting)
+	if err != nil {
+		return fmt.Errorf("dns backup: %w", err)
 	}
-	return backupAndWriteFile(resolvConfHeadFile, resolvConfHeadBackupFile, setting)
-}
 
-func restoreSystemDNSServer() {
-	restoreFileIfExists(resolvConfBackupFile, resolvConfFile)
-	restoreFileIfExists(resolvConfHeadBackupFile, resolvConfHeadFile)
+	err = backupAndWriteFile(resolvConfHeadFile, resolvConfHeadBackupFile, setting)
+	if err != nil {
+		return fmt.Errorf("dns head backup: %w", err)
```

Comment:
Similar to above.

```suggestion
		return err
```

Reply:
Also if we are simply returning the `err` here, we can just do `return backupAndWriteFile(...)` directly.